k8s.go

```diff
@@ -17,16 +17,24 @@ const (
 	defaultK8sResourcePolicy = "keep"
 )
 
-type k8sSecretOptions struct {
-	Namespace     string `long:"namespace" description:"The Kubernetes namespace the secret is located in"`
-	SecretName    string `long:"secret-name" description:"The name of the Kubernetes secret"`
-	SecretKeyName string `long:"secret-key-name" description:"The name of the key/entry within the secret"`
-	Base64        bool   `long:"base64" description:"Encode as base64 when storing and decode as base64 when reading"`
+type k8sObjectType string
+
+const (
+	ObjectTypeSecret    k8sObjectType = "Secret"
+	ObjectTypeConfigMap k8sObjectType = "ConfigMap"
+)
+
+type k8sObjectOptions struct {
```

Comment:
If the rename of the CLI flag to the more general `--k8s.name` is not preferred, we could just make a separate `k8sConfigmapOptions` struct.

Reply:
Okay, I updated things so that we can avoid renaming any CLI flags ✅ Still did some translating from the secret/configmap specific structs to this common "object" struct prior to passing to (save/read)k8s. Let me know if this is any better!